### ISSUE TITLE
Yoda comparisons not flagged against functions

### ISSUE THREAD

**Issue by soderlind** [WordPress-Coding-Standards/WordPress-Coding-Standards#652]
For code consistency, shouldn't `has_the_force() === true` be flagged ?:

```php
if ( has_the_force() === true ) {
    $victorious = you_will( $be );
}
```

**Comment by westonruter** [comment]
I recall that the yoda condition sniff is working by identifying when there is a variable used as an lvalue. Yoda condition checks are really only important when variables (lvalues) are used in the comparison because both `true = has_the_force()` and `has_the_force() = true` would be syntax errors.

**Comment by GaryJones** [comment]
From the [Handbook](https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions) (my emphasis):

> When doing logical comparisons, always put the variable on the right side, constants or **literals on the left**.

There's nothing in there about it _only_ applying when the logical comparison involves a variable.

So while it may not actually matter in terms of bug prevention, I'd argue that the sniff doesn't actually check for all possible cases that the current Handbook wording alludes to. It's clear, that somewhere there's an ambiguity that means some folks think `some_function() === true` should be flagged.

**Comment by jrfnl** [comment]
> > When doing logical comparisons, always put the variable on the right side, constants or literals on the left.

> There's nothing in there about it only applying when the logical comparison involves a variable.

But there is also nothing in that handbook sentence which indicates that this rule should be applied to a logical comparison involving a function call (in contrast to a variable, constant or literal)....

**Comment by westonruter** [comment]
We can update the handbook to be more clear here.

**Comment by GaryJones** [comment]
> We can update the handbook to be more clear here.

How about:

"When doing logical comparisons involving variables, always put the variable on the right side, constants or literals on the left. If neither side is a variable, the order does not matter."

**Comment by soderlind** [comment]
So, instead of consistency you rewrite the documentation ? 😕

**Comment by GaryJones** [comment]
If it removes ambiguity in the purpose of having the check, yes. There's nothing stopping you from having `true == some_function()` in your code if you wish - I probably will as that's what I personally prefer - but it isn't at risk of creating a potential bug if one of the `=` is missing.

**Comment by westonruter** [comment]
I updated the docs as follows:

> When doing logical comparisons involving variables, always put the variable on the right side and put constants, literals, or function calls on the left side. If neither side is a variable, the order is not important. (In <a href="https://en.wikipedia.org/wiki/Value_(computer_science)#Assignment:_l-values_and_r-values">computer science terms</a>, in comparisons always try to put l-values on the right and r-values on the left.)

https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/#yoda-conditions

**Comment by jrfnl** [comment]
@westonruter Is there any way to keep track of the changes made to the handbook ? (other than the very minimal changelog at the bottom)

**Comment by GaryJones** [comment]
Follow / Watch in the top right. Generates an email when changes are made.

**Comment by jrfnl** [comment]
@GaryJones Thanks!
